tests/simulation/test_start_simulator.py

```diff
@@ -0,0 +1,183 @@
+        plots,
+        plot_size,
+    ) = start_simulator_main(True, root_path=chia_root)
+    await kwargs["peer_api"].bt.setup_keys(fingerprint=fingerprint, reward_ph=farming_puzzle_hash)
+    await kwargs["peer_api"].bt.setup_plots(
+        num_og_plots=plots, num_pool_plots=0, num_non_keychain_plots=0, plot_size=plot_size
+    )
+    service = Service(**kwargs)
+
+    await service.start()
+
+    yield service._api
+
+    service.stop()
+    await service.wait_closed()
+
+
+async def get_num_coins_for_ph(simulator_client: SimulatorFullNodeRpcClient, ph: bytes32) -> int:
+    return len(await simulator_client.get_coin_records_by_puzzle_hash(ph))
+
+
+class TestStartSimulator:
+    """
+    These tests are designed to test the user facing functionality of the simulator.
+    """
+
+    @pytest_asyncio.fixture(scope="function")
+    async def get_user_simulator(
+        self,
+    ) -> AsyncGenerator[Tuple[FullNodeSimulator, Path, Dict[str, Any], str, int], None]:
+        # Create and setup temporary chia directories.
+        chia_root = Path(tempfile.TemporaryDirectory().name)
+        mnemonic, fingerprint = mnemonic_fingerprint()
+        config = create_config(chia_root, fingerprint)
+        lockfile = singleton(daemon_launch_lock_path(chia_root))
+        crt_path = chia_root / config["daemon_ssl"]["private_crt"]
+        key_path = chia_root / config["daemon_ssl"]["private_key"]
+        ca_crt_path = chia_root / config["private_ssl_ca"]["crt"]
+        ca_key_path = chia_root / config["private_ssl_ca"]["key"]
+        assert lockfile is not None
+        shutdown_event = asyncio.Event()
+        ws_server = WebSocketServer(chia_root, ca_crt_path, ca_key_path, crt_path, key_path, shutdown_event)
+        await ws_server.start()  # type: ignore[no-untyped-call]
+
+        async for simulator in start_simulator(chia_root):
+            yield simulator, chia_root, config, mnemonic, fingerprint
+
+        await ws_server.stop()
+
+    @pytest.mark.asyncio
+    async def test_start_simulator(
+        self, get_user_simulator: Tuple[FullNodeSimulator, Path, Dict[str, Any], str, int]
+    ) -> None:
+        simulator, root_path, config, mnemonic, fingerprint = get_user_simulator
+        ph_1 = get_puzzle_hash_from_key(fingerprint, key_id=1)
+        ph_2 = get_puzzle_hash_from_key(fingerprint, key_id=2)
+        dummy_hash = std_hash(b"test")
+        num_blocks = 2
+        # connect to rpc
+        rpc_port = config["full_node"]["rpc_port"]
+        simulator_rpc_client = await SimulatorFullNodeRpcClient.create(
+            config["self_hostname"], uint16(rpc_port), root_path, config
+        )
+        # test auto_farm logic
+        assert await simulator_rpc_client.get_auto_farming()
+        await time_out_assert(10, simulator_rpc_client.set_auto_farming, False, False)
+        await simulator.autofarm_transaction(dummy_hash)  # this should do nothing
+        await asyncio.sleep(3)  # wait for block to be processed
```

Comment:
Sleep is bad and causes flaky tests. You can do a timeout assert to wait for a specific block height

Reply:
It's not supposed to advance, the sleepyness is only for sanity checking. Ex: fail condition

Reply:
got it thanks